### DISCUSSION TITLE
Add linting configuration to docs

### DISCUSSION THREAD

**Discussion by nuno-andre** [pydantic/pydantic#4686] [Feature Request]
There are some specificities of Pydantic that do not go well with some linters. 

E.g. _flake8_ complains about `validator`, which can be easily addressed with:

```ini
[flake8]
classmethod-decorators =
    classmethod
    validator
```

I propose to open a (_Usage_ &rarr; _Usage with linters_) section in the documentation to collect these settings.

**Comment by samuelcolvin** [comment]
Sounds good. create on `main` and we'll include it in the new docs.
